cmd/siklu/common_boot.c

```diff
@@ -64,21 +88,28 @@ int load_kernel_image(void) {
 	char formatted_bootargs[1024];
 	const char *old_bootargs;
 	unsigned int fdt_addr;
-
-	if (strict_strtoul(dtb_load_address(), 16, &fdt_addr) < 0)
-		return -EINVAL;
-
-	const char* fdt_param = siklu_fdt_getprop_string(fdt_addr, "/chosen", "bootargs", NULL);
-
-	if (!IS_ERR(fdt_param)) {
-		old_bootargs = env_get("bootargs");
-		snprintf(formatted_bootargs, sizeof(formatted_bootargs), "%s %s", old_bootargs, fdt_param);
-		printf("SIKLU BOOT: Added DTS-specific bootargs: %s\n", fdt_param);
-		env_set("bootargs", formatted_bootargs);
+	char *fdt_file;
+	char *vendor;
+
+	if (is_fit_image()) {
+		fdt_file = env_get("fdtfile");
+		if (fdt_file == NULL) {
+			printk(KERN_ERR "Could not find FDT file environment variable");
+			return -EINVAL;
+		}
+		vendor = env_get("fdtvendor");
```

Comment:
I would suggest to use a define instead of a variable. environment variables can be removed and therefore make the device unbootable.